### ISSUE TITLE
[Megatron-SWIFT][RM] Reward model eval accuracy stays 1.0 after swapping chosen/rejected, even with synthetic contradictory probe

### ISSUE THREAD

**Issue by Morgansgun** [modelscope/ms-swift#9370]
### Checklist / 检查清单

- [x] I have searched existing issues, and this is a new bug report. / 我已经搜索过现有的 issues，确认这是一个新的 bug report。

### Bug Description / Bug 描述

Hi ms-swift team,

I am training a reward model with megatron rlhf --rlhf_type rm on a Qwen3.6-35B-A3B-like MoE model. I encountered a very strange issue during RM evaluation.

The training/eval logs show that RM accuracy quickly reaches 1.0 after only around 10–20 steps. However, when I swap chosen and rejected in the validation set, the eval accuracy is still 1.0. This is unexpected because the pairwise RM accuracy should drop significantly, ideally close to 0 if all pairs are reversed.

Due to data privacy, I cannot share the real dataset. But the issue can also be reproduced with a synthetic contradictory probe.

### How to Reproduce / 如何复现

I used swift 4.0.1 and flashattention 0.4.2

My RM data follows the standard messages + rejected_response format:
{
  "messages": [
    {"role": "user", "content": "..."},
    {"role": "assistant", "content": "chosen response"}
  ],
  "rejected_response": "rejected response"
}

I have confirmed that the swapped validation set correctly swaps，so this does not seem to be caused by an incorrect swap script.

The original training command is roughly:
megatron rlhf \
  --rlhf_type rm \
  --model <BASE_MODEL> \
  --dataset <TRAIN_JSONL> \
  --val_dataset <EVAL_JSONL> \
  --packing true \
  --tuner_type lora \
  --lora_rank 8 \
  --lora_alpha 32 \
  --target_modules all-linear \
  --tensor_model_parallel_size 2 \
  --pipeline_model_parallel_size 2 \
  --expert_model_parallel_size 2 \
  --micro_batch_size 1 \
  --global_batch_size 32 \
  --recompute_granularity full \
  --recompute_method uniform \
  --recompute_num_layers 2 \
  --optimizer_cpu_offload true \
  --num_train_epochs 1 \
  --finetune true \
  --lr 1e-4 \
  --lr_warmup_fraction 0.05 \
  --min_lr 1e-5 \
  --max_length 16384 \
  --packing_length 16384 \
  --padding_free true \
  --sequence_parallel true \
  --attention_backend flash \
  --center_rewards_coefficient 0.01 \
  --save_safetensors true \
  --merge_lora true

For evaluation, I used a command like:（actually I don't know how to evaluate the RM by swift, could you answer this question?）
megatron rlhf \
  --rlhf_type rm \
  --model <MERGED_RM_MODEL> \
  --dataset <DUMMY_TRAIN_JSONL> \
  --val_dataset <EVAL_JSONL_OR_SWAPPED_EVAL_JSONL> \
  --load_from_cache_file false \
  --packing false \
  --padding_free false \
  --tensor_model_parallel_size 1 \
  --pipeline_model_parallel_size 4 \
  --expert_model_parallel_size 2 \
  --micro_batch_size 1 \
  --global_batch_size 32 \
  --finetune true \
  --lr 1e-10 \
  --min_lr 1e-10 \
  --lr_warmup_fraction 0.0 \
  --num_train_epochs 1 \
  --eval_steps 1 \
  --max_length 4096 \
  --dataset_num_proc 1 \
  --dataloader_num_workers 0 \
  --sequence_parallel false \
  --attention_backend flash \
  --center_rewards_coefficient 0.01

### Additional Information / 补充信息

Could you please help confirm the following?

1. In megatron rlhf --rlhf_type rm, is the reported eval accuracy computed strictly as: reward_chosen > reward_rejected. Using the raw scalar reward from the value head?
2. Are chosen_reward and rejected_reward in the logs raw model scores for prompt + chosen and prompt + rejected, or are they transformed values after pairwise loss processing. 
3. Is there an official recommended way to run RM-only evaluation or inference for a Megatron-SWIFT trained reward model, preferably one that outputs per-sample.
4. Where should I add debug prints to inspect the tokenized chosen and rejected inputs before forward, so I can verify whether the same answer gets identical input_ids when it appears in the chosen branch and rejected branch?
5. 
Thanks for your help. This issue is important because the RM eval accuracy looks perfect, but the swapped-set behavior suggests that the current evaluation result may not reflect true pairwise reward ranking.
